scripts/core/editor3/components/handlePastedText.js

```diff
@@ -116,10 +116,26 @@ function processPastedHtml(props, html) {
         contentState = Modifier.setBlockType(contentState, selection, 'atomic');
     }
 
-    contentState = Modifier.replaceWithFragment(contentState, selection, OrderedSet(blocks));
-    contentState = sanitizeContent(contentState, acceptedInlineStyles);
+    let nextEditorState = EditorState.push(
+        editorState,
+        Modifier.replaceWithFragment(contentState, selection, OrderedSet(blocks)),
```

Comment:
If 'selection' contains the first block the data for this block will be overwrite.

Reply:
I see there's already a PR submitted by you which fixes this, so I'll wait for it to get merged and will rebase this branch to include it.